pi/test_heartbeat.py

```diff
@@ -57,47 +61,41 @@ def test_log_ip_when_darkstat_html_contains_flic_mac_address(self):
 class Test__parse_link_quality_from_iwconfig_output(object):
 
     def test_link_quality_1(self):
-        with open(os.path.dirname(__file__) + '/test_files/sample_iwconfig_output/link_quality_1.txt', 'r') as iwconfig_output_file:
-            with unittest.mock.patch('heartbeat.logging') as mock_logging:
-                iwconfig_output = iwconfig_output_file.read()
-                heartbeat.parse_link_quality_from_iwconfig_output(iwconfig_output)
-                mock_logging.info.assert_called_once_with('wlan0 link quality is %f', 1.0)
+        with open(os.path.dirname(__file__) + '/test_files/sample_iwconfig_output/link_quality_1.txt', 'r') as output_file, patch('heartbeat.logging') as mock_logging:
```

Comment:
That's cool. I didn't realize that you could combine the "with"s like this.

Reply:
yeah, I thought I should clean things up a bit.